### PR TITLE
Staking UI

### PR DESCRIPTION
**Pull request by LeifuChen**
## Description
- [x] Add kwenta value icon component
- [x] Add epoch label
- [x] Update `dark`/`light` theme
- [x] Update `en.json`
- [x] Add escrow table row
- [x]  Add escrow table row `select` and show the total amount of `select`
- [x] Add epoch switcher
- [x] Integrate the mock data
- [x] Integrate the contract
- [X] Calculate APY
- [X] Add vesting entries
- [x] Add Claimable rewards
- [x] Add Vestable Qty
- [x] Add Fee/Immediate Vestable Qty
- [x] Add Stake/Unstake function
- [x] Add Redeem function
- [x] Update mobile

### CODE REVIEW

File: translations/en.json [hunk 1]
```diff
@@ -491,6 +491,67 @@
 				"open-positions": "Open Positions"
 			}
 		},
+		"stake": {
+			"portfolio": {
+				"title": "Portfolio",
```
Comment (platschi) [inline translations/en.json:496]: Might consider renaming this to "Staking", "Staking Portfolio" or "Staking Dashboard" or "Staking Overview" or "KWENTA Staking" or "$Kwenta Portfolio" or so, the term _Portfolio_ alone would be confusing to me here given we already have "Portfolio" on the Dashboard Overview. Also it's not immediately clear *what* we're staking here :D

Prob best to discuss with some English-native CC :)

**Review by platschi — COMMENTED** [review]
Looks good already and works pretty smooth. A few general layout/UI comments before testing functionality:

- [ ] On OP Goerli `Redemption`, frontend recognizes my `5 vKwenta` as both `v`Kwenta and `ve`Kwenta, even though I do not have `ve`Kwenta in the wallet:

Screenshot 2022-10-28 at 10 06 12

- [ ] Tooltips: As a start, can you add tooltips from the tooltip component to the titles of each of the six boxes in the top Portfolio? I will provide copy soon. Feel they are necessary to explain in short what is what
- [ ] Missing a short one sentence help text with a link to the staking docs to explain the staking mechanism and all its intricacies, we should consider adding this somewhere
- [ ] Lots of empty space between the top portfolio part and the Staking tabs, consider lowering the empty space? See dashboard overview for comparison

**Comment by platschi** [comment]
### Trading Rewards
- Remove `Estimated rewards` and `Estimated fee share` (calc takes too long)
- Remove `Claimable rewards: Epoch 1` and just show "Total Claimable Rewards"
- Remove Claim Epoch 1 button and just show "Claim all"
- After claiming my trading rewards, the 'Claim all' button did not disable, so I was able to do a second transaction (which failed)

### Escrow
- Escrow: Rename `eKWENTA` to `escrowed Kwenta`

### General
- remove the word `Current` from `Current Balance` (in Stake/Unstake boxes)

File: translations/en.json [hunk 2]
```diff
@@ -429,7 +429,7 @@
 			"rewards": "Rewards",
 			"markets": "Markets",
 			"governance": "Governance",
-			"staking": "Staking"
+			"stake": "Stake"
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply translations/en.json]: Yes. The change has been made and the file name has also been changed.

File: translations/en.json (hunk 1 above)
Comment (LeifuChen) [reply translations/en.json]: It has been renamed.

File: sections/dashboard/Stake/StakingTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	margin-bottom: 4px;
+	align-items: center;
+	visibility: ${(props) => (props.active ? 'visible' : 'hidden')};
+	border-radius: ${(props) => (props.isRounded ? '50px' : '8px')};
+	transition-duration: 0s;
+
+	${media.lessThan('md')`
+		background: none;
+		flex-direction: column;
+		color: #b1b1b1;
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply sections/dashboard/Stake/StakingTabs.tsx]: Fixed.

File: sections/dashboard/Stake/InputCards/EscrowInputCard.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+		if (unstakeTxn?.hash) {
+			monitorTransaction({
+				txHash: unstakeTxn?.hash,
+				onTxConfirmed: () => {
+					setAmount('');
+				},
+			});
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [unstakeTxn?.hash]);
```
Comment (koredefashokun) [inline sections/dashboard/Stake/InputCards/EscrowInputCard.tsx:112]: Same here as well.

File: sections/dashboard/Stake/EscrowTable.tsx [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+			monitorTransaction({
+				txHash: vestTxn?.hash,
+				onTxConfirmed: () => {
+					setCheckedState(new Array(data.length).fill(false));
+					setCheckAllState(false);
+				},
+			});
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [vestTxn?.hash]);
```
Comment (koredefashokun) [inline sections/dashboard/Stake/EscrowTable.tsx:83]: Same here.

File: sections/dashboard/Stake/EscrowTable.tsx [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+
+	const selectAll = () => {
+		if (checkAllState) {
+			setCheckedState(new Array(data.length).fill(false));
+			setCheckAllState(false);
+		} else {
+			setCheckedState(new Array(data.length).fill(true));
+			setCheckAllState(true);
+		}
+	};
```
Comment (koredefashokun) [inline sections/dashboard/Stake/EscrowTable.tsx:37]: Maybe we should use a useCallback here.

**Review by koredefashokun — COMMENTED** [review]
Looks pretty good. Just left a couple of places where we can improve code quality.

File: sections/dashboard/Stake/StakingTabs.tsx [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+				startDate,
+				endDate,
+				label,
+			});
+			setPeriod(i);
+			setStart(epochStart ?? 0);
+			setEnd(epochEnd ?? 0);
+			setCurrentEpochLabel(label);
+		});
+		return epochData;
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply sections/dashboard/Stake/StakingTabs.tsx:70]: This is much better. Thanks.

File: sections/dashboard/Stake/EscrowTable.tsx (hunk 6 above)
Comment (LeifuChen) [reply sections/dashboard/Stake/EscrowTable.tsx:37]: Update. Thanks.

File: sections/dashboard/Stake/StakingTab.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+	const { data, write: getReward } = useContractWrite(getRewardConfig);
+
+	useEffect(() => {
+		if (data?.hash) {
+			monitorTransaction({
+				txHash: data?.hash,
+			});
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [data?.hash]);
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply sections/dashboard/Stake/StakingTab.tsx]: Thanks for pointing out this. I will also refactor this to use typechain contracts later.

File: sections/dashboard/Stake/InputCards/StakeInputCard.tsx [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+		if (unstakeTxn?.hash) {
+			monitorTransaction({
+				txHash: unstakeTxn?.hash,
+				onTxConfirmed: () => {
+					setAmount('');
+				},
+			});
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [unstakeTxn?.hash]);
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply sections/dashboard/Stake/InputCards/StakeInputCard.tsx]: Thanks. The change ha been made.

File: sections/dashboard/Stake/InputCards/RedeemInputCard.tsx [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+	}, [veApproveTxn?.hash]);
+
+	useEffect(() => {
+		if (veRedeemTxn?.hash) {
+			monitorTransaction({
+				txHash: veRedeemTxn?.hash,
+			});
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [veRedeemTxn?.hash]);
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply sections/dashboard/Stake/InputCards/RedeemInputCard.tsx]: Thanks. The change ha been made.

File: sections/dashboard/Stake/InputCards/EscrowInputCard.tsx (hunk 4 above)
Comment (LeifuChen) [reply sections/dashboard/Stake/InputCards/EscrowInputCard.tsx]: Thanks. The change has been made.

File: sections/dashboard/Stake/EscrowTable.tsx (hunk 5 above)
Comment (LeifuChen) [reply sections/dashboard/Stake/EscrowTable.tsx]: Thanks. The change has been made.

File: sections/dashboard/Stake/TradingRewardsTab.tsx [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+			fileNames.push(`trading-rewards-snapshots/epoch-${i - 1}.json`);
+		});
+		return fileNames;
+	}, [periods]);
+
+	const allEpochQuery = useGetFiles(fileNames);
+	const allEpochData = useMemo(() => allEpochQuery?.data ?? [], [allEpochQuery?.data]);
+
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	let rewards: [any, any, any, any, any][] = [];
```
[the remark this answers is not in the document]
Comment (LeifuChen) [reply sections/dashboard/Stake/TradingRewardsTab.tsx:60]: Yes. Thanks for pointing out this. I will refactor this one later.

**Review by platschi — APPROVED** [review]

**Pull request merged**